### PR TITLE
Gopkg.*,pkg/generator/templates.go: update to kubernetes 1.11.2

### PR DESCRIPTION
**Pull request by wongma7**
this continues https://github.com/operator-framework/operator-sdk/pull/391 @hasbro17 PTAL

### CODE REVIEW

**Comment by wongma7** [comment]
memcached_test.go:223: Building operator docker image
        	memcached_test.go:226: error: exit status 1
        		Command Output: Error: failed to build: (building memcached-operator...
        		vendor/github.com/operator-framework/operator-sdk/pkg/k8sclient/client.go:31:2: cannot find package "k8s.io/client-go/restmapper" in any of:
        			/home/travis/gopath/src/github.com/example-inc/memcached-operator/vendor/k8s.io/client-go/restmapper (vendor tree)
        			/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/k8s.io/client-go/restmapper (from $GOROOT)
        			/home/travis/gopath/src/k8s.io/client-go/restmapper (from $GOPATH)
        		)

**Comment by shawn-hurley** [comment]
@wongma7 @hasbro17 I think the test is failing because the generated operator is not using this branches SDK code.

What is the best way to either make the generated code point to this branch for the test or do we ignore e2e tests? 

IMO we should consider updating the tests, to always pull the branch that is being tested of the sdk. But I don't know if we should block this PR? 

/cc @AlexNPavel

**Comment by AlexNPavel** [comment]
This should be using our code, not sure why it is failing: https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/memcached_test.go#L72
I'll look at this more in depth later and see if I can figure out what's going on

**Comment by hasbro17** [comment]
@shawn-hurley The e2e test actually does make sure that for the generated operator the current PR branch of the SDK is vendored.
https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/memcached_test.go#L72-L73

The problem we're doing this post `dep ensure` which ran against the master branch of the SDK which does have a dependency on `k8s.io/client-go/restmapper`. So that's never added to the vendor.

If the initial `dep ensure` ran against this PR then `k8s.io/client-go/restmapper` would be a transitive dependency of `pkg/sdk` and would be added to the vendor.

It's always always a bit tricky when we update pkg/sdk that introduces new transitive dependencies.

**Comment by hasbro17** [comment]
I think we can fix how we run this test and have it run `dep ensure` against this PR and then get the correct transitive dependencies.

Instead of just replacing the vendored SDK directory directly after running `dep ensure`, we can have the test get the Git-SHA or revision of the current PR and update the test-operator's Gopkg.toml to use the current PR's revision.

```sh
PR_REVISION=git rev-parse HEAD
sed -i 's|branch = "master"|revision = "$PR_REVISION"|g' Gopkg.toml

# Run dep ensure to update the vendored SDK and its transitive deps with the current PR 
dep ensure
```

**Comment by shawn-hurley** [comment]
I think you're going to have to change the dep source as well, depending on travis variables, But I think this approach will be less error-prone IMO

**Comment by AlexNPavel** [comment]
Due to the way that dep works, we cannot fix this with dep (remote references like the ones github/travis use are not supported). We simple have to merge this and then check if travis's master build works.

**Comment by hasbro17** [comment]
Travis exposes the following envs:
 `TRAVIS_PULL_REQUEST_SHA`  gives us the HEAD of the current PR. We use this as the revision.
`TRAVIS_PULL_REQUEST_SLUG ` gives us the `owner_name/repo_name` of the fork of the PR. We'll use this as the [Source](https://golang.github.io/dep/docs/Gopkg.toml.html#source) in for the sdk constraint. 

So we can update the Gopkg.toml accordingly in our test to vendor the PR commit.

**Comment by AlexNPavel** [comment]
New PR open with fix ^

**Comment by hasbro17** [comment]
@wongma7 Can you please rebase your PR from master. With #525 we should be able to properly test your changes.

**Comment by hasbro17** [comment]
The test is still failing but this time the operator is built and run successfully. Its just not reconciling correctly as we expect it to (i.e create a memcached deployment for a memcached CR).

@wongma7 Can you do a manual test with the example on the README/user-guide to test out if an operator with your changes can still watch and use the client correctly. I'm going to verify that myself.

File: pkg/k8sclient/client.go [hunk 1]
```diff
@@ -95,31 +96,28 @@ func (c *resourceClientFactory) GetResourceClient(apiVersion, kind, namespace st
...
-		Name:       mapping.Resource,
-		Namespaced: mapping.Scope == meta.RESTScopeNamespace,
-		Kind:       gvk.Kind,
+	gvr := &schema.GroupVersionResource{
+		Group:    gvk.Group,
+		Version:  gvk.Version,
+		Resource: mapping.Resource.String(),
 	}
-	return resource, nil
+	return gvr, nil
```
Comment (hasbro17) [inline pkg/k8sclient/client.go]: After testing it manually I found the issue.
`mapping.Resource` is of type `schema.GroupVersionResource` and not of type string.

Replace lines 115-120:
```Go
gvr := &schema.GroupVersionResource{
		Group:    gvk.Group,
		Version:  gvk.Version,
		Resource: mapping.Resource.String(),
	}
	return gvr, nil
```
with:
```Go
return &mapping.Resource, nil
```

Comment (wongma7) [reply pkg/k8sclient/client.go]: thanks, fixed, works for me locally as well now

**Comment by shawn-hurley** [comment]
LGTM Thank you!

**Comment by AlexNPavel** [comment]
LGTM

**Comment by hasbro17** [comment]
LGTM

**Pull request merged**